Modules/Core/Transform/include/itkMatrixOffsetTransformBase.hxx

```diff
@@ -99,7 +99,9 @@ MatrixOffsetTransformBase<TParametersValueType, VInputDimension, VOutputDimensio
   m_MatrixMTime.Modified();
   m_Offset.Fill(OutputVectorValueType{});
   m_Translation.Fill(OutputVectorValueType{});
-  m_Center.Fill(InputPointValueType{});
+  // Fixed parameters must be preserved when setting the transform to identity
+  // the center is part of the stationary fixed parameters
+  // and should not be modified by SetIdentity. m_Center.Fill(InputPointValueType{});
```

Comment:
Thanks very much for giving this approach a try, Hans! I think it's the right way to go. As far as I can see, the Scale*Transform tests still pass 👍 (Right? Please check!) Only [itkSimilarity2DTransformTest](https://open.cdash.org/test/1454097934) appears to fail, but I think itkSimilarity2DTransformTest itself is wrong! I'll have another look.

Reply:
@N-Dekker Thanks. I won't be able to spend time digging into this until after next week.  I appreciate any assistance I can get.

2877 - itkSimilarity2DTransformTest (Failed)

Reply:
Technically, just adding `transform->SetCenter({})` to line 159 of itkSimilarity2DTransformTest.cxx should make the CI happy.

Conceptually, people should no longer assume that `transform->SetIdentity()` "automagically" resets the center. It's a breaking change. But I think it's a correct one.